backend/ipsec/handle_charon.go

```diff
@@ -0,0 +1,185 @@
+// Copyright 2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ipsec
+
+import (
+	"fmt"
+	"os/exec"
+	"time"
+
+	"github.com/coreos/flannel/Godeps/_workspace/src/github.com/bronze1man/goStrongswanVici"
+	log "github.com/coreos/flannel/Godeps/_workspace/src/github.com/golang/glog"
+	"github.com/coreos/flannel/subnet"
+)
+
+//TODO how to set debug level with standalone charon
+//TODO update depending on how we package
+const (
+	configFilePath = "/etc/strongswan.conf"
+	charonPath     = "/usr/local/libexec/ipsec/charon"
+	rekeyTime      = "10m"
+)
+
+type CharonIKEDaemon struct {
+	path           string
+	configFilePath string
+}
+
+func NewCharonIKEDaemon() (*CharonIKEDaemon, error) {
+	path, err := exec.LookPath(charonPath)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Info("Charon path: ", path)
+
+	return &CharonIKEDaemon{
+		path:           path,
+		configFilePath: configFilePath,
+	}, nil
+}
+
+func (charon *CharonIKEDaemon) RunCharonIKEDaemon() error {
```

Comment:
stylistic nit: I would rename func name to `RunDaemon` or maybe even better to just `Run`